src/Document.php

```diff
@@ -87,4 +92,15 @@ public function __construct($data = [], $options = [])
             $this->clean();
         }
     }
+
+    /**
+     * Returns the Elastica\Result object that can be used for getting
+     * extra information about the document, such as the score and highlights.
```

Comment:
They could return the relevant empty value. If we add this accessor, callers can't chain methods as the elasticResult() will sometimes return null.